src/applications/vaos/new-appointment/redux/reducer.js

```diff
@@ -245,10 +222,32 @@ export default function formReducer(state = initialState, action) {
           [action.pageKey]: 'home',
         };
       }
+      let newPages = state.pages;
+      let newData = action.data || state.data;
+
+      if (
+        getTypeOfCare(newData)?.id !== getTypeOfCare(state.data)?.id &&
+        (state.pages.vaFacility || state.data.vaFacility)
+      ) {
+        newPages = unset('vaFacility', newPages);
+        newData = unset('vaFacility', newData);
+      }
+
+      // reset community care provider if type of care changes
+      if (
+        getTypeOfCare(newData)?.id !== getTypeOfCare(state.data)?.id &&
+        (state.pages.ccPreferences || !!state.data.communityCareProvider?.id)
+      ) {
+        newPages = unset('ccPreferences', newPages);
+        newData = set('communityCareProvider', {}, newData);
+      }
+
       return {
         ...state,
         pageChangeInProgress: true,
         previousPages: updatedPreviousPages,
+        data: newData,
```

Comment:
Update the form data here every time the page changes.

Also maybe worth noting that using the back and forward buttons will not persist your form data changes, which I don't think is necessarily bad. We could account for that if we need to, though.